### ISSUE TITLE
Unsupported arithmetic operations should raise a `NotImplementedError`

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/sunpy#6640]
### Describe the bug

They currently raise a `NotImplemented`. See the discussion on sunpy/ndcube#541.

### To Reproduce

import sunpy

sunpy.map.Map(...)
etc

**Comment by ayshih** [comment]
Probably just miscommunication, but if you are referring to unsupported arithmetic operations given the types of the two operands, those should properly return `NotImplemented` – as they currently do – and not `NotImplementedError`.

**Comment by wtbarnes** [comment]
Yes, given the subsequent discussion on the previously linked PR, I agree. The two approaches are now consistent.
